app/lib/methods/loadMessagesForRoom.ts

```diff
@@ -5,29 +5,37 @@ import log from '../../utils/log';
 import { getMessageById } from '../database/services/Message';
 import { generateLoadMoreId } from '../utils';
 import updateMessages from './updateMessages';
+import { IRocketChat } from '../../definitions/IRocketChat';
+import { IMessage, IRoom, SubscriptionType, TMessageModel } from '../../definitions';
+import sdk from '../rocketchat/services/sdk';
+import RocketChat from '../rocketchat';
 
 const COUNT = 50;
 
-async function load({ rid: roomId, latest, t }) {
-	let params = { roomId, count: COUNT };
+async function load({ rid: roomId, latest, t }: Pick<IRoom, 'rid' | 'latest' | 't'>) {
+	let params = { roomId, count: COUNT } as { roomId: string; count: number; latest: string };
 	if (latest) {
 		params = { ...params, latest: new Date(latest).toISOString() };
 	}
 
-	const apiType = this.roomTypeToApiType(t);
+	const apiType = RocketChat.roomTypeToApiType(t);
 	if (!apiType) {
 		return [];
 	}
 
 	// RC 0.48.0
-	const data = await this.sdk.get(`${apiType}.history`, params);
+	// @ts-ignore
+	const data: any = await sdk.get(`${apiType}.history`, params);
 	if (!data || data.status === 'error') {
 		return [];
 	}
 	return data.messages;
 }
 
-export default function loadMessagesForRoom(args) {
+export default function loadMessagesForRoom(
+	this: IRocketChat,
```

Comment:
Here you don't need `this`, since you don't use `this` anymore in the other function.
You can call `load` directly without being `.call`.

Reply:
cool! thanks!